server/static/dashboard/live.js

```diff
@@ -0,0 +1,141 @@
+var connection = new window.sharedb.Connection(new WebSocket(shareURL));
+
+var collabs = connection.createSubscribeQuery('collabs', {
+  project: project,
+  $sort: { '_m.ctime': -1 }
+}, {});
+
+collabs.on('ready', function() { insertCollabs(collabs.results, 0); });
+collabs.on('insert', insertCollabs);
+
+var searchbox = document.querySelector('#search');
+var countdisplay = document.querySelector('#matches');
+
+var searchregex = undefined;
+searchbox.addEventListener('input', function() {
+  searchregex = undefined;
+  if (searchbox.value.length > 0) {
+    try {
+      searchregex = new RegExp(searchbox.value, 'i');
+    } catch (e) {}
+  }
+});
+
+function insertCollabs(collabs, atIndex) {
+  var list = document.querySelector('#collabs');
+  collabs.forEach(function(collab, idx) {
+    var item = document.importNode(document.querySelector('#collab').content, true);
+    var root = item.querySelector('.collab');
+    root.setAttribute('id', 'collab-' + collab.id);
+    let users = collab.data.users.slice().sort();
+    root.dataset.users = users.join(',');
+    var link = item.querySelector('a');
+    var href = '/dashboard/' + project + '/' + collab.id;
+    link.setAttribute('href', href);
+    link.textContent = users.join('\n');
+    list.insertBefore(item, list.children[atIndex + idx]);
+
+    // TODO: move into view file
+    var codeline = document.createElement('span');
+    codeline.style.fontSize = '8px';
+    codeline.style.background = 'rgba(252, 248, 227)';
+    root.querySelector('.panel-body').appendChild(document.createElement('br'));
+    root.querySelector('.panel-body').appendChild(codeline);
+
+    connection.createFetchQuery('files', { collabid: collab.id }, {}, function(err, files) {
+      if (err) { throw err; }
+
+      var typingtimeoutid = undefined;
+      function updateTypingDisplay() {
+        root.classList.add('typing');
+        clearTimeout(typingtimeoutid);
+        typingtimeoutid = setTimeout(function() {
+          root.classList.remove('typing');
+        }, 1000);
+      }
+
+      function updateErrorDisplay() {
+        var maxlevel = 0;
+        files.forEach(function(file) {
+          for (var k in file.data.markers) {
+            file.data.markers[k].forEach(function(marker) {
+              maxlevel = Math.max(maxlevel, marker.severity);
+            });
+          }
+        });
+        root.classList.remove('error');
```

Comment:
Have an array of statuses, `[ ok, warning, error ]` and use that to both remove classes and add back the class.